### PR TITLE
Add CI to run tests

### PR DESCRIPTION
**Pull request by boarnoah**

### CODE REVIEW

**Comment by boarnoah** [comment]
Hm, that summary isn't the greatest eh. like the wording and stuff

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Build unit tests project
+        run: |
+          mkdir build
+          cd build
+          cmake -DCMAKE_BUILD_TYPE=Release ..
+          cmake --build . --
```
Comment (right-mouse) [inline .github/workflows/ci.yml]: If you're not passing in any args to `build` (such as `-j12` etc.) you can omit the trailing `--`.

**Comment by right-mouse** [comment]
Is it possible to have the tests results show up at the top of the README?

**Comment by boarnoah** [comment]
> Is it possible to have the tests results show up at the top of the README?

You mean badge for default branch's status? 


~~EDIT: Doesn't look like, I'm not too sure how badges work but presumably they need something to poll re: current status.~~

~~Github actions has the concept of [Environments](https://docs.github.com/en/actions/reference/environments) and at work for example we use that to quickly get an idea re: if an environment deployed.~~

~~I suppose that it could also be used to show failed if there was a failure in tests (by making successful test pass required for the rest of "deploying" to an environment succeed).~~

**Comment by boarnoah** [comment]
As we found out, you can click on a github action and choose to generate a test badge and use that in readmes etc..

**Review by right-mouse — APPROVED** [review]
Looks good to me.

**Review by aashkir — CHANGES_REQUESTED** [review]
Looks good, but it should be called: Continuous, not Continous (On line 1)

**Comment by boarnoah** [comment]
Mouse pointed out in chat he would like to see int he future separate badges for windows vs linux statuses, with github actions that likely means separate workflows (since 1 workflow = 1 badge).

When/if we do that then this would be renamed to CI linux etc.. to make that fact clear

**Review by aashkir — APPROVED** [review]
Nice work.

**Pull request merged**